File: TME.Store/TME.Store.Domain/Models/Product.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TME.Store.Domain.Models
{
    public class Product
    {
        public string Symbol { get; private set; }
        public string Producer { get; private set; }
        public string Description { get; private set; }
        public string Category { get; private set; }
        public string Unit { get; private set; }
        public string Photo { get; private set; }
        public string Thumbnail { get; private set; }
        public int Amount { get; private set; }     
        public ProductPrice ProductPrice { get; }

        public Product(string symbol,
                        string producer,
                        string description,
                        string category,
                        string unit,
                        string thumbnail,
                        string photo,
                        int amount,
                        ProductPrice price
                       )
        {
            Symbol = symbol;
            Producer = producer;
            Description = description;
            Category = category;
            Unit = unit;
            ProductPrice = price;
            Photo = "https:" + photo;
            Thumbnail = "https:" + thumbnail;
            Amount = amount;
        }
    }
}

File: tme_store/lib/models/product.dart
import 'package:tme_store/api/models/api_poduct.dart';
import 'package:tme_store/api/models/api_product_price_with_stock.dart';
import 'package:tme_store/models/price_orffer.dart';
import 'package:tme_store/models/product_price.dart';

class Product {
  final String symbol;

  final String producer;

  final String description;

  final String category;

  final String photoUrl;

  final String thumbnailUrl;

  final double weight;

  final int amount;

  final String unit;

  final ProductPrice price;

  Product(
      {this.symbol,
      this.producer,
      this.description,
      this.category,
      this.photoUrl,
      this.thumbnailUrl,
      this.weight,
      this.amount,
      this.unit,
      this.price});

  static Product fromApiModels(
      ApiProduct apiProduct, ApiProductPriceWithStock productPrice) {
    return Product(
        symbol: apiProduct.symbol,
        category: apiProduct.category,
        description: apiProduct.description,
        producer: apiProduct.producer,
        weight: apiProduct.weight,
        unit: apiProduct.unit,
        photoUrl: apiProduct.photoUrl,
        thumbnailUrl: apiProduct.thumbnailUrl,
        amount: productPrice.amount,
        price: ProductPrice(
            vatRate: productPrice.vatRate,
            vatType: productPrice.vatType,
            priceOffers: productPrice.prices
                .map((apiPrice) => PriceOffer(
                    ammountInOffer: apiPrice.amount,
                    isSpecialPrice: apiPrice.isSpecial,
                    price: apiPrice.value))
                .toList()));
  }
}
